packages/pages/src/vite-plugin/build/closeBundle/moduleLoader.ts

```diff
@@ -25,7 +25,12 @@ export const loadTemplateModules = async (
     }
 
     const templateModuleInternal =
-      convertTemplateModuleToTemplateModuleInternal(p, templateModule, true);
+      convertTemplateModuleToTemplateModuleInternal(
+        p,
+        templateModule,
+        true,
+        false // doesn't matter here, it's already been validated at this point
```

Comment:
not a huge fan of this assumption here or elsewhere

Reply:
Same. I didn't want to have to do this everywhere though:
```
const clientServerRenderTemplates = getGlobalClientServerRenderTemplates(
    projectStructure.templatesRoot,
    projectStructure.scopedTemplatesPath
  );
```

The reason this flag is necessary is because loadTemplateModules and convertTemplateModuleToTemplateModuleInternal happen in a variety of places and validation happens during load. This is beneficial because it checks that you don't have `getHeadConfig` defined if you're using custom rendering templates (because those should set the `<head>`, `<title>`, etc tags instead.

I could move the validation out of here and do it somewhere else like during build, but it's nice for the user to get feedback immediately during local dev. Any ideas or thoughts on if we should move it? I could just do the actual determination of the custom rendering templates per above in all of these assumption places, but it will mean also updating a bunch of params that need to be passed through.

Reply:
is it possible to pass templateModuleInternal around rather than converting the template module? 

We're doing this on build start and then again when closeBundle is called. I wonder if we could persist that state. That gets around the problem a bit

Reply:
We're only loading templates in closeBundle. I updated one place where I had "doesn't matter" and then also removed moduleLoader altogether because it was actually unused (and added back the one validation it had directly into closeBundle).